python/shotgun_model/shotgun_model.py

```diff
@@ -841,6 +865,62 @@ def _load_external_data(self):
         """
         pass
 
+    def _get_additional_columns(self, item, is_leaf, columns):
```

Comment:
I ran a test with 10 columns on a model with 100,000 rows and there wasn't any noticeable timing difference between showing the view with or without those columns.  I think we're alright for the kind of calculations we're doing (just pulling values from the SG_DATA dict), although maybe for more complicated calculations this could be an issue.

We talked about it and'll move forward as is for now.